inspire_utils/date.py

```diff
@@ -99,7 +99,15 @@ def loads(cls, string):
             ValueError: month must be in 1..12
 
         """
-        parts = (int(part) for part in string.split('-'))
+
+        date_parts = string.split('-')
+
+        if len(date_parts) == 2 and len(date_parts[1]) < 2:
+            raise ValueError('Month must be in MM format')
+        if len(date_parts) == 3 and len(date_parts[2]) < 2:
```

Comment:
there's another case that you're not catching, where there's a full date but only the month is missing a leading 0, like in `2015-1-15`.